### DISCUSSION TITLE
Ask me about a very strange IOS problem

### DISCUSSION THREAD

**Discussion by AsaminOgawa** [strongswan/strongswan#2018] [Q&A]
I reinstalled a vpn server on centos 8 with the same configuration as before, but the previous one could connect and now I can only connect to WIN10 and MACOS. Neither iPad nor iPhone can connect. The configuration file and log file have been given. , I hope someone with good intentions can help.

[root@VPN conf.d]# cat test.conf
authorities {
    lets-encrypt {
        file = /etc/strongswan/swanctl/x509ca/ca.pem
    }
}

connections {
    ikev2 {
        version = 2
        local_addrs = %any
        remote_addrs = %any
        proposals = 3des-aes128-aes192-aes256-sha1-sha256-sha384-modp1024, aes256gcm16-sha384-ecp384, aes256-sha256-modp2048
        dpd_delay = 30s
        send_certreq = no
        send_cert = always
        rekey_time = 0s
        pools = pool-ipv4
        local {
            certs = /etc/strongswan/swanctl/x509/server.pem
            id = vpn.example.com
        }
        remote {
            eap_id = %any
            auth = eap-radius
        }
        children {
            ikev2 {
                esp_proposals = aes256-aes128-3des-des-null-sha1, aes256gcm16-sha384-modp2048,aes256-sha256-modp2048
                local_ts = 0.0.0.0/0
                rekey_time = 0s
            }
        }
    }
}

secrets {
    private-1 {
        file = /etc/strongswan/swanctl/private/privkey.pem
    }
}

pools {
    pool-ipv4 {
        addrs = 10.31.3.100/25
        dns = 10.31.3.1
    }
}

log:
[root@VPN conf.d]# swanctl --log
plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
15[NET] received packet: from xxx.xxx.xxx.xxx[20030] to 10.31.3.2[500] (370 bytes)
15[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(REDIR_SUP) N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) ]
15[IKE] xxx.xxx.xxx.xxx is initiating an IKE_SA
15[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
15[IKE] local host is behind NAT, sending keep alives
15[IKE] remote host is behind NAT
15[IKE] DH group ECP_256 unacceptable, requesting MODP_2048
15[ENC] generating IKE_SA_INIT response 0 [ N(INVAL_KE) ]
15[NET] sending packet: from 10.31.3.2[500] to xxx.xxx.xxx.xxx[20030] (38 bytes)
16[NET] received packet: from xxx.xxx.xxx.xxx[20030] to 10.31.3.2[500] (562 bytes)
16[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(REDIR_SUP) N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) ]
16[IKE] xxx.xxx.xxx.xxx is initiating an IKE_SA
16[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
16[IKE] local host is behind NAT, sending keep alives
16[IKE] remote host is behind NAT
16[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
16[NET] sending packet: from 10.31.3.2[500] to xxx.xxx.xxx.xxx[20030] (472 bytes)
07[NET] received packet: from xxx.xxx.xxx.xxx[20060] to 10.31.3.2[4500] (464 bytes)
07[ENC] unknown attribute type INTERNAL_DNS_DOMAIN
07[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR MASK DHCP DNS ADDR6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) N(NON_FIRST_FRAG) SA TSi TSr N(MOBIKE_SUP) ]
07[CFG] looking for peer configs matching 10.31.3.2[vpn.example.com]...xxx.xxx.xxx.xxx[10.93.16.111]
07[CFG] selected peer config 'ikev2'
07[IKE] initiating EAP_IDENTITY method (id 0x00)
07[IKE] peer supports MOBIKE
07[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
07[IKE] authentication of 'vpn.example.com' (myself) with ECDSA_WITH_SHA256_DER successful
07[IKE] sending end entity cert "CN=vpn.example.com"
07[IKE] sending issuer cert "C=US, O=Let's Encrypt, CN=R3"
07[ENC] generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
07[ENC] splitting IKE message (2576 bytes) into 3 fragments
07[ENC] generating IKE_AUTH response 1 [ EF(1/3) ]
07[ENC] generating IKE_AUTH response 1 [ EF(2/3) ]
07[ENC] generating IKE_AUTH response 1 [ EF(3/3) ]
07[NET] sending packet: from 10.31.3.2[4500] to xxx.xxx.xxx.xxx[20060] (1236 bytes)
07[NET] sending packet: from 10.31.3.2[4500] to xxx.xxx.xxx.xxx[20060] (1236 bytes)
07[NET] sending packet: from 10.31.3.2[4500] to xxx.xxx.xxx.xxx[20060] (260 bytes)

**Comment by tobiasbrunner** [comment]
The iOS client apparently doesn't seem to like the server certificate, which sounds similar to #1986.

**Reply by AsaminOgawa** [reply]
Thank you very much for your answer. I am using letencrypt's free SSL certificate, but I can use it with the same configuration and the same certificate before, but I deleted the previous virtual machine.

**Comment by AsaminOgawa** [comment]
I saw that someone configured it successfully. I used his configuration encryption protocol and still got an error. I listed the configuration table and errors below.

[root@VPN ~]# cat /etc/strongswan/swanctl/conf.d/test.conf
authorities {
    lets-encrypt {
        file = /etc/strongswan/swanctl/x509ca/ca.pem
    }
}

connections {
    ikev2 {
        version = 2
        local_addrs = %any
        remote_addrs = %any
        #proposals = 3des-aes128-aes192-aes256-sha1-sha256-sha384-modp1024, aes256gcm16-sha384-ecp384, aes256-aes192-sha256-modp3072-modp2048
        proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-ecp521,aes192-sha256-modp3072,aes256gcm16-sha384-prfsha384-ecp384,aes256-aes128-sha256-sha1-modp3072-modp2048-modp1024,default
        dpd_delay = 30s
        send_certreq = no
        send_cert = always
        rekey_time = 0s
        pools = pool-ipv4
        local {
            certs = /etc/strongswan/swanctl/x509/server.pem
            id = vpn.example.com
        }
        remote {
            eap_id = %any
            auth = eap-radius
        }
        children {
            ikev2 {
                #esp_proposals = aes256-aes128-3des-des-null-sha1, aes256gcm16-sha384-modp2048, aes256-sha256, aes256-sha256-modp2048
                esp_proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-modp3072,aes192-sha256-ecp256-modp3072,aes256-aes128-sha256-sha1-modp3072-modp2048-modp1024,default
                local_ts = 0.0.0.0/0
                rekey_time = 0s
            }
        }
    }
}

secrets {
    private-1 {
        file = /etc/strongswan/swanctl/private/privkey.pem
    }
}

pools {
    pool-ipv4 {
        addrs = 10.31.3.100/25
        dns = 10.31.3.1
    }
}

13[NET] received packet: from 10.31.3.1[500] to 10.31.3.2[500] (370 bytes)
13[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(REDIR_SUP) N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) ]
13[IKE] 10.31.3.1 is initiating an IKE_SA
13[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
13[IKE] local host is behind NAT, sending keep alives
13[IKE] remote host is behind NAT
13[IKE] DH group ECP_256 unacceptable, requesting MODP_2048
13[ENC] generating IKE_SA_INIT response 0 [ N(INVAL_KE) ]
13[NET] sending packet: from 10.31.3.2[500] to 10.31.3.1[500] (38 bytes)
13[NET] received packet: from 10.31.3.1[500] to 10.31.3.2[500] (562 bytes)
13[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(REDIR_SUP) N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) ]
13[IKE] 10.31.3.1 is initiating an IKE_SA
13[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
13[IKE] local host is behind NAT, sending keep alives
13[IKE] remote host is behind NAT
13[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
13[NET] sending packet: from 10.31.3.2[500] to 10.31.3.1[500] (472 bytes)
08[NET] received packet: from 10.31.3.1[4500] to 10.31.3.2[4500] (464 bytes)
08[ENC] unknown attribute type INTERNAL_DNS_DOMAIN
08[ENC] parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR MASK DHCP DNS ADDR6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) N(NON_FIRST_FRAG) SA TSi TSr N(MOBIKE_SUP) ]
08[CFG] looking for peer configs matching 10.31.3.2[vpn.example.com]...10.31.3.1[192.168.0.101]
08[CFG] selected peer config 'ikev2'
08[IKE] initiating EAP_IDENTITY method (id 0x00)
08[IKE] peer supports MOBIKE
08[IKE] received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
08[IKE] authentication of 'vpn.example.com' (myself) with ECDSA_WITH_SHA256_DER successful
08[IKE] sending end entity cert "CN=vpn.example.com"
08[IKE] sending issuer cert "C=US, O=Let's Encrypt, CN=R3"
08[ENC] generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
08[ENC] splitting IKE message (2576 bytes) into 3 fragments
08[ENC] generating IKE_AUTH response 1 [ EF(1/3) ]
08[ENC] generating IKE_AUTH response 1 [ EF(2/3) ]
08[ENC] generating IKE_AUTH response 1 [ EF(3/3) ]
08[NET] sending packet: from 10.31.3.2[4500] to 10.31.3.1[4500] (1236 bytes)
08[NET] sending packet: from 10.31.3.2[4500] to 10.31.3.1[4500] (1236 bytes)
08[NET] sending packet: from 10.31.3.2[4500] to 10.31.3.1[4500] (260 bytes)

**Comment by AsaminOgawa** [comment]
Let me talk about my solution now. It is indeed caused by the certificate problem. Use -k 4096 in the acme.sh script and do not generate the default 2048 certificate.

**Reply by tobiasbrunner** [reply]
Just to clarify, you did not use an RSA 2048 certificate before but an ECDSA certificate (different algorithm), and as I suspected in #1986 already that could be a problem if the client doesn't support this type of certificate.

**Comment by focusexplorer** [comment]
I solve the probrem by looking at the log of iphone, found that the iphone verify server certificate failed，when I installed CA certificate in iphone. It became ok
